motioneye/uploadservices.py

```diff
@@ -14,28 +14,33 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import StringIO
+import datetime
 import ftplib
 import json
 import logging
 import mimetypes
 import os
 import os.path
-import StringIO
 import time
 import urllib
 import urllib2
+from abc import ABCMeta, abstractmethod
+from inspect import isabstract
+
 import pycurl
+import webdav3.client
 
 import settings
 import utils
-import config
-import datetime
 
 _STATE_FILE_NAME = 'uploadservices.json'
 _services = None
 
 
-class UploadService(object):
+class UploadService:
```

Comment:
Needs to test other services as UploadService and GoogleBase has changed.

Reply:
What do you mean by test? Unit test, or just click testing through the UI?

Reply:
I mean regression test. As base classes that existing services are based on have changed, they (existing services) need to be tested as well to make sure there was no unwanted side effect on them.